course_discovery/apps/course_metadata/search_indexes.py

```diff
@@ -410,6 +418,37 @@ def prepare_search_card_display(self, obj):
 
             return []
         return [degree.search_card_ranking, degree.search_card_cost, degree.search_card_courses]
+    
+    def prepare_num_of_courses(self, obj):
+        return obj.courses.count()
+    
+    def prepare_banner_image_url(self, obj):
+        if obj.banner_image_url:
+            return obj.banner_image_url
+        elif obj.banner_image:
+            return obj.banner_image.url
+        return None
+    
+    def prepare_bundle_price(self, obj):
+        bundle_price = 0
+        course_runs = list(obj.course_runs)
+        for course_run in course_runs:
+            first_enrollable_paid_seat_price = course_run.first_enrollable_paid_seat_price
+            if first_enrollable_paid_seat_price:
+                bundle_price += first_enrollable_paid_seat_price
+        return bundle_price
+    
+    def prepare_bundle_currency(self, obj):
+        course_runs = list(obj.course_runs)
+        if course_runs:
+            first_course_run = course_runs[0]
+            first_seat = first_course_run.seats.first()
+            if first_seat:
+                return first_seat.currency.code
+        return None
```

Comment:
Can we do this in reverse order. Unhappy path first.  e.g None should be returned first.

Reply:
Changed